MoviesTVSentiments/app/src/main/java/com/google/moviestvsentiments/service/account/AccountRepository.java

```diff
@@ -116,4 +116,24 @@ void setIsCurrent(String name, boolean isCurrent) {
             accountDao.setIsCurrent(name, isCurrent);
         });
     }
+
+    /**
+     * Sends the list of pending Accounts to the server. If the server successfully adds them to its
+     * database, then the Accounts are updated locally to no longer be pending.
+     */
+    public void syncPendingAccounts() {
+        executor.execute(() -> {
+            List<Account> pendingAccounts = accountDao.getPendingAccounts();
+            if (pendingAccounts.isEmpty()) {
+                return;
+            }
+
+            ApiResponse<List<Account>> response = webService.syncPendingAccounts(pendingAccounts);
+            if (response.isSuccessful()) {
```

Comment:
Wonder if failures need to trigger some kind of retry?

Currently I see this is only triggered when network becomes available. If the sync after the network comes available fails, does that next sync happens at the next network disconnect/connect?

Something you can consider for scheduling periodic work is WorkManager (https://developer.android.com/topic/libraries/architecture/workmanager). Just a suggestion, I am sure you can do it in other ways.

Reply:
Yes, if the sync fails, then the next sync will happen at the next network reconnect or when the app is closed and reopened.

Reply:
I see. I feel next app open is fine for now.

If you want to improve this later, maybe add a TODO here? And once you add the sync for other things, the scheduling sync or retry can be added for all of them? What do you think.

Reply:
Sure, that sounds like a good plan.

Reply:
Feel free to add an issue for tracking purpose.